src/devnet/external-server.ts

```diff
@@ -121,11 +131,18 @@ export abstract class ExternalServer {
                 this.childProcess = null;
                 if (code !== 0 && isAbnormalExit) {
                     if (this.connected) {
-                        const msg = `${this.processName} spawned and connected successfully, but later exited with code=${code}`;
+                        const msg = logger.isFile(this.stderr)
+                            ? `${this.processName} spawned and connected successfully, but later exited with code=${code}\nError logged to file ${this.stderr}`
+                            : `${this.processName} spawned and connected successfully, but later exited with code=${code}`;
                         throw new HardhatPluginError(PLUGIN_NAME, msg);
                     } else {
-                        const msg = `${this.processName} connect exited with code=${code}:\n${this.lastError}`;
-                        reject(new HardhatPluginError(PLUGIN_NAME, msg));
+                        let msg = logger.isFile(this.stderr)
+                            ? `integrated-devnet connect exited with code=${code}:\nError logged to file ${this.stderr}`
+                            : `integrated-devnet connect exited with code=${code}`;
+                        if (!this.stderr) {
```

Comment:
Shouldn't this also be applied up in the case when `this.connected` is `true`?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/179/commits/f350dcbe0fe1f66ead3d695f85a4e7608772cc94